### PR TITLE
Documentation for v1.6.16 and Kaldi Support

### PR DESCRIPTION
**Pull request by LexiconCode**
# Documentation for v1.6.16 and Kaldi Support

## Description

This pull request updates the documentation from the caster v1.0.0 rewrite to v1.6.16. It includes changes to Caster to support Kaldi speech recognition engine.

View the updated updated [documentation](https://caster-lexiconcode.readthedocs.io/en/documentation/) on my personal branch readthedocs.

I'm going to need help from the community reviewing the documentation and providing feedback. Including but not limited to the following:

- Typos and grammar
- Does the documentation make sense?
- Does the user feel like they're being guided in a way that makes sense as a progress from the beginning to the end of the documentation?
- and anything else

### Speech Recognition Engines

Support for Kaldi has been added.

A breaking change is how engines are loaded. The old method with how natllink was configured meant that were run on the same process with Dragon. This means that Dragon has to restart in order for Caster to reload python modules. The new method to install and launch engines are out of prrocess meaning the grammar is run on their own thread. 

For Dragon Naturally users this is significant. This means you can restart Caster and not restart Dragon speech recognition engine. It takes the restart time to two seconds or so. Overall the install process and running Caster has changed In the following ways.

- The install documentation now has dedicated Install .bat per engine.
  - Configuring natlink is no longer required as long as it's installed.

- Caster now launches through .bat for all engines.
  - Caster no longer launches automatically with DNS
    - **Note**: Users can choose to keep the in process method of running Dragon and Natlink.
    - Command prompt window replaces Natlink messaging window (A replacement is in the works)
  - WSR no longer launches when clicking on _caster.py use the .bat file instead

### Command Changes

- Unified reboot command `reboot caster` that restarts Caster. Only unless you are using the Dragon in process method.
- Renamed Caster Settings editor GUI command from `Launch settings window` to  `caster settings editor`

### Documentation

This is broad overview answers to many tiny details to describe here.

- Updated all examples to use the new Caster 1.x.x loading method
- Structure documentation to guide the user
  - Choosing a speech recognition engine and installing Caster
  - Using Caster using commands and features
  - Creating and customizing your own commands with Dragonfly and Caster
  - Explaining Caster settings and user directory
- Added a FAQ
- Deleted old documentation referencing features that have long since been removed
- Re-factored some of the location of the files in the directory and on ReadTheDocs.

**ToDo**

- ~~Address leftover `ToDo`~~ 
- ~~Cleanup typos~~ 
- ~~Address feedback~~ 
- ~~Test all rules examples to make sure their valid in documentation.~~

**ToDo  Long-term**: Not in this pull request

- Add and update the technical details of how the underlying changes from the rewrite work with Caster.
- Add more command CCR chains examples for programming by voice
- Videos for related content
- Hardware and Microphone Recommendations
- Debugging Rules

## Related Issue

Overhaul _caster.py #638
Old Docs #741

## Motivation and Context

Caster even before the rewrite contained a lot of out of date documentation much of which has been removed. Now the documentation references the new features contained in the rewrite.

## How Has This Been Tested

- I've tested running all supported engines through their respective `.bat` files

- Tested the reboot commands for WSR/DNS # Waiting on a fix for Kaldi Python 3.8.1

- [Importing the win32ui](https://github.com/dictation-toolbox/dragonfly/issues/205) module crashes Natlink if used out of process. win32ui has been removed from the Caster code base and its functionality reimplemment it with dragonflies cross-platform functions.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue or bug)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Renamed existing command phrases (we discourage this without a strong rationale).

## Checklist

- [x] I have read the CONTRIBUTING document.
- [x] My code follows the code style of this project.
- [x] I have checked that my code does not duplicate functionality elsewhere in Caster.
- [x] I have checked for and utilized existing command phrases from within Caster (delete if not applicable). 
- [ ] My code implements all the features I wish to merge in this pull request.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

## Maintainer/Reviewer Checklist

- [x] Basic functionality has been tested and works as claimed.
- [x] New documentation is clear and complete.
- [ ] Code is clear and readable.

### CODE REVIEW

**Comment by LexiconCode** [comment]
I need to redo the commit history. First replacing files that already exist so that the Diff between them can be seen. Then re-factor as a separate commit.

File: docs/readthedocs/Customize_Caster/Customizing_Starter_Rules.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+<<<SPEC>>>
+shock -> earthquake
+<<<NOT_SPECS>>>
+sauce -> up
+dunce -> down
+lease -> left
+ross -> right
+```
+
+​	This `words.txt` will create a rule filter which goes through **all** Caster rules at boot time and replaces the word "shock" in any spec with the word "earthquake". It will also replace "sauce", "dunce" , "lease" and "ross"  with "up", "down", "left" . and "right" in extras and defaults in any rule. The triple-angle-brackets indicate mode changes. Valid modes are `SPEC` (for specs only), `EXTRA`, `DEFAULT`, `NOT_SPECS` (for extras and defaults, but not specs), and `ANY` (for specs, extras, and defaults). The default mode is `ANY`.
```
Comment (MarkRx) [inline docs/readthedocs/Customize_Caster/Customizing_Starter_Rules.md]: What exactly are SPEC, EXTRA, DEFAULT, NOT_SPECS? An end user might not know what those terms mean or how they relate to modifying words.txt

Comment (LexiconCode) [reply docs/readthedocs/Customize_Caster/Customizing_Starter_Rules.md]: How well does this link address some of those [concerns](https://caster-lexiconcode.readthedocs.io/en/documentation/readthedocs/Rule_Construction/Taxonomy_of_a_Rule/)?

Comment (MarkRx) [reply docs/readthedocs/Customize_Caster/Customizing_Starter_Rules.md]: Ah yes that explains it thanks

File: docs/readthedocs/Caster_Settings/rules.md [hunk 2]
```diff
@@ -1,30 +1,32 @@
 ## Rules
 
-All rules managed by Caster are tracked in this config file `rules.toml `located in located in `AppData\Local\caster\settings`. This configuration file allows you not only to see what grammars are `active` but `whitelist` which grammars even get loaded by Caster during startup.
+All rules managed by Caster are tracked in this config file `rules.toml `located in located in `AppData\Local\caster\settings`. This configuration file allows you not only to see what rules are active but change with the whitelist which rules are loaded by Caster during startup.
+
+Rules can exist in different states:
+
+- `Loaded ` During startup Caster loads all available rules on startup into the speech recognition engine. If a rule is not loaded it cannot be `enabled`/`disabled` and the rules commands commands cannot be recognized.
```
Comment (qubitz) [inline docs/readthedocs/Caster_Settings/rules.md]: rules commands commands cannot be recognized.

**Pull request merged**